internal/ingest/ingest.go

```diff
@@ -173,7 +197,7 @@ func (li *legIngester) Sync(ctx context.Context, peerID peer.ID, opts ...SyncOpt
 	// Start syncing. Notifications for the finished
 	// sync will be done asynchronously.
 	log.Debugf("Started syncing process with provider %s", sub)
-	watcher, cncl, err := sub.ls.Sync(ctx, peerID, c)
+	watcher, cncl, err := sub.ls.Sync(ctx, peerID, c, li.advSelectorNode)
```

Comment:
Discussed in 1:1 meetings. Summary of approach: this PR introduces a selector that would sync an advertisement _and_ its linked chunks, a similar behaviour to how it worked before selector was introduced. This means we do not need to re-engineer the storage hook mechanism in storetheindex link system. Instead, approach that in future PRs along with other opportunities for optimisation, e.g. batch processing of advertisements instead of one-by-one.